src/main/java/org/jboss/remoting3/remote/IntIndexHashMap.java

```diff
@@ -200,14 +200,18 @@ public boolean add(final V v) {
     }
 
     public <T> T[] toArray(final T[] a) {
-        final ArrayList<T> list = new ArrayList<T>(size());
-        list.addAll((Collection<T>) this);
+        final ArrayList<V> list = new ArrayList<V>(size());
```

Comment:
I see no test failures now.  This PR resolves the StackOverflow that was mentioned that had been seen when running the 3.3 branch tests.  It also resolves the regression of the rejected exception that was being logged.  The org.jboss.remoting3.test.racecondition.CloseConnectingEndpointTestCase which I saw fail on my box happened because I had BYTEMAN_HOME set, removing that all tests pass.